pantab/_reader.py

```diff
@@ -75,49 +75,75 @@ def _read_table(*, connection: tab_api.Connection, table: TableType) -> pd.DataF
 
 
 def frame_from_hyper(
-    database: Union[str, pathlib.Path],
+    source: Union[str, pathlib.Path, tab_api.Connection],
     *,
     table: TableType,
     hyper_process: Optional[tab_api.HyperProcess] = None,
 ) -> pd.DataFrame:
     """See api.rst for documentation"""
 
-    with tempfile.TemporaryDirectory() as tmp_dir, ensure_hyper_process(
-        hyper_process
-    ) as hpe:
-        tmp_db = shutil.copy(database, tmp_dir)
-        with tab_api.Connection(hpe.endpoint, tmp_db) as connection:
-            return _read_table(connection=connection, table=table)
+    if isinstance(source, tab_api.Connection):
+        assert (
```

Comment:
Rather than assert can you change this to a `ValueError`? Also can you add a test that we raise this error when both are provided?

Reply:
done